helpers/api/BatchManager.test.ts

```diff
@@ -0,0 +1,247 @@
+import { expect } from 'chai'
+import sinon from 'sinon'
+
+import { BatchCache } from './BatchCache'
+import { BatchManager, Request } from './BatchManager'
+
+const mockConnection = 'https://www.goingnowhere.com'
+
+describe('BatchManager', () => {
+  let clock: sinon.SinonFakeTimers
+
+  beforeEach(function () {
+    clock = sinon.useFakeTimers()
+  })
+
+  afterEach(function () {
+    clock.restore()
+  })
+
+  it('should batch call into single request', async () => {
+    const mockBatch = createMockBatch()
+    const mockFetchJson = sinon.stub().resolves([]) as any
+
+    const batchManager = createBatchManager(mockFetchJson)
+    await batchManager.batchCall(mockBatch)
+
+    expect(mockFetchJson).to.have.been.calledWith(mockConnection, JSON.stringify(mockBatch))
+  })
+
+  it('should exclude cache hits', async () => {
+    const mockBatch = createMockBatch()
+    const mockFetchJson = sinon.stub().resolves(
+      mockBatch.map(() => ({
+        result: 'cached-result',
+      })),
+    ) as any
+
+    const batchManager = createBatchManager(mockFetchJson)
+
+    // First call
+    await batchManager.batchCall(mockBatch)
+    mockFetchJson.reset()
+
+    // Second call
+    await batchManager.batchCall(mockBatch)
+
+    expect(mockFetchJson).to.not.have.been.called
```

Comment:
```suggestion
    expect(mockFetchJson).to.have.been.called.once
```
and remove the `reset` above?